### ISSUE TITLE
Error: Cannot find module '@babel/runtime/helpers/interopRequireDefault'

### ISSUE THREAD

**Issue by rafakwolf** [okta/okta-auth-js#459]
**Steps I did:**
Removed node_modules and installed all packages again.

**My environment:**
❯ node --version
v12.15.0

Also, I'm using typescript 4 as a transpiler only, so my source code files are javascript.

Here's the stack-trace:

```
'Error: Cannot find module '@babel/runtime/helpers/interopRequireDefault'
Require stack:
- /home/wolf/dev/liferay-cloud-api/api/node_modules/@okta/okta-auth-js/cjs/server/index.js
- /home/wolf/dev/liferay-cloud-api/api/src/manager/UserManager.js
- /home/wolf/dev/liferay-cloud-api/api/src/manager/UserManagerFactory.js
- /home/wolf/dev/liferay-cloud-api/api/test/env/seed.js
- /home/wolf/dev/liferay-cloud-api/api/test/unit/versions.js
- /home/wolf/dev/liferay-cloud-api/api/node_modules/gulp-mocha/node_modules/mocha/lib/mocha.js
- /home/wolf/dev/liferay-cloud-api/api/node_modules/gulp-mocha/node_modules/mocha/lib/cli/one-and-dones.js
- /home/wolf/dev/liferay-cloud-api/api/node_modules/gulp-mocha/node_modules/mocha/lib/cli/options.js
- /home/wolf/dev/liferay-cloud-api/api/node_modules/gulp-mocha/node_modules/mocha/bin/mocha
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:793:17)
    at Function.Module._load (internal/modules/cjs/loader.js:686:27)
    at Module.require (internal/modules/cjs/loader.js:848:19)
    at require (internal/modules/cjs/helpers.js:74:18)
    at Object.<anonymous> (/home/wolf/dev/liferay-cloud-api/api/node_modules/@okta/okta-auth-js/cjs/server/index.js:3:30)
    at Module._compile (internal/modules/cjs/loader.js:955:30)
    at Module._extensions..js (internal/modules/cjs/loader.js:991:10)
    at Object.require.extensions.<computed> [as .js] (/home/wolf/dev/liferay-cloud-api/api/node_modules/ts-node/src/index.ts:851:44)
    at Module.load (internal/modules/cjs/loader.js:811:32)
    at Function.Module._load (internal/modules/cjs/loader.js:723:14)
    at Module.require (internal/modules/cjs/loader.js:848:19)
    at require (internal/modules/cjs/helpers.js:74:18)
    at Object.<anonymous> (/home/wolf/dev/liferay-cloud-api/api/src/manager/UserManager.js:11:1)
    at Module._compile (internal/modules/cjs/loader.js:955:30)
    at Module.m._compile (/home/wolf/dev/liferay-cloud-api/api/node_modules/ts-node/src/index.ts:858:23)
    at Module._extensions..js (internal/modules/cjs/loader.js:991:10)
    at Object.require.extensions.<computed> [as .js] (/home/wolf/dev/liferay-cloud-api/api/node_modules/ts-node/src/index.ts:861:12)
    at Module.load (internal/modules/cjs/loader.js:811:32)
    at Function.Module._load (internal/modules/cjs/loader.js:723:14)
    at Module.require (internal/modules/cjs/loader.js:848:19)
    at require (internal/modules/cjs/helpers.js:74:18)
    at Object.<anonymous> (/home/wolf/dev/liferay-cloud-api/api/src/manager/UserManagerFactory.js:1:1)
    at Module._compile (internal/modules/cjs/loader.js:955:30)
    at Module.m._compile (/home/wolf/dev/liferay-cloud-api/api/node_modules/ts-node/src/index.ts:858:23)
    at Module._extensions..js (internal/modules/cjs/loader.js:991:10)
    at Object.require.extensions.<computed> [as .js] (/home/wolf/dev/liferay-cloud-api/api/node_modules/ts-node/src/index.ts:861:12)
    at Module.load (internal/modules/cjs/loader.js:811:32)
    at Function.Module._load (internal/modules/cjs/loader.js:723:14)
    at Module.require (internal/modules/cjs/loader.js:848:19)
    at require (internal/modules/cjs/helpers.js:74:18)
    at Object.<anonymous> (/home/wolf/dev/liferay-cloud-api/api/test/env/seed.js:8:1)
    at Module._compile (internal/modules/cjs/loader.js:955:30)
    at Module.m._compile (/home/wolf/dev/liferay-cloud-api/api/node_modules/ts-node/src/index.ts:858:23)
    at Module._extensions..js (internal/modules/cjs/loader.js:991:10)
    at Object.require.extensions.<computed> [as .js] (/home/wolf/dev/liferay-cloud-api/api/node_modules/ts-node/src/index.ts:861:12)
    at Module.load (internal/modules/cjs/loader.js:811:32)
    at Function.Module._load (internal/modules/cjs/loader.js:723:14)
    at Module.require (internal/modules/cjs/loader.js:848:19)
    at require (internal/modules/cjs/helpers.js:74:18)
    at Object.<anonymous> (/home/wolf/dev/liferay-cloud-api/api/test/unit/versions.js:1:1)
    at Module._compile (internal/modules/cjs/loader.js:955:30)
    at Module.m._compile (/home/wolf/dev/liferay-cloud-api/api/node_modules/ts-node/src/index.ts:858:23)
    at Module._extensions..js (internal/modules/cjs/loader.js:991:10)
    at Object.require.extensions.<computed> [as .js] (/home/wolf/dev/liferay-cloud-api/api/node_modules/ts-node/src/index.ts:861:12)
    at Module.load (internal/modules/cjs/loader.js:811:32)
    at Function.Module._load (internal/modules/cjs/loader.js:723:14)
    at Module.require (internal/modules/cjs/loader.js:848:19)
    at require (internal/modules/cjs/helpers.js:74:18)
    at /home/wolf/dev/liferay-cloud-api/api/node_modules/gulp-mocha/node_modules/mocha/lib/mocha.js:334:36
    at Array.forEach (<anonymous>)
    at Mocha.loadFiles (/home/wolf/dev/liferay-cloud-api/api/node_modules/gulp-mocha/node_modules/mocha/lib/mocha.js:331:14)
    at Mocha.run (/home/wolf/dev/liferay-cloud-api/api/node_modules/gulp-mocha/node_modules/mocha/lib/mocha.js:809:10)
    at Object.exports.singleRun (/home/wolf/dev/liferay-cloud-api/api/node_modules/gulp-mocha/node_modules/mocha/lib/cli/run-helpers.js:108:16)
    at exports.runMocha (/home/wolf/dev/liferay-cloud-api/api/node_modules/gulp-mocha/node_modules/mocha/lib/cli/run-helpers.js:142:13)
    at Object.exports.handler (/home/wolf/dev/liferay-cloud-api/api/node_modules/gulp-mocha/node_modules/mocha/lib/cli/run.js:292:3)
    at Object.runCommand (/home/wolf/dev/liferay-cloud-api/api/node_modules/gulp-mocha/node_modules/yargs/lib/command.js:242:26)
    at Object.parseArgs [as _parseArgs] (/home/wolf/dev/liferay-cloud-api/api/node_modules/gulp-mocha/node_modules/yargs/yargs.js:1096:28)
    at Object.parse (/home/wolf/dev/liferay-cloud-api/api/node_modules/gulp-mocha/node_modules/yargs/yargs.js:575:25)
    at Object.exports.main (/home/wolf/dev/liferay-cloud-api/api/node_modules/gulp-mocha/node_modules/mocha/lib/cli/cli.js:68:6)
    at Object.<anonymous> (/home/wolf/dev/liferay-cloud-api/api/node_modules/gulp-mocha/node_modules/mocha/bin/mocha:162:29)
    at Module._compile (internal/modules/cjs/loader.js:955:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:991:10)
    at Module.load (internal/modules/cjs/loader.js:811:32)
    at Function.Module._load (internal/modules/cjs/loader.js:723:14)
    at Function.Module.runMain (internal/modules/cjs/loader.js:1043:10)
    at internal/main/run_main_module.js:17:11
```

**Comment by shuowu** [comment]
@rafakwolf Does the following thread help? https://github.com/babel/babel/issues/9265

**Comment by rafakwolf** [comment]
hey @shuowu ... unfortunately, it doesn't :|     
I also tried to install  _@babel/runtime_  package ... then I faced the error:
```
/home/wolf/dev/liferay-cloud-api/api/node_modules/@okta/okta-auth-js/lib/oauthUtil.js:15
import http from './http';
^^^^^^
SyntaxError: Cannot use import statement outside a module
``` 

just one more piece of info... I'm using the okta-auth-js in a nodejs backend

**Comment by rafakwolf** [comment]
Hey @shuowu ... I was importing this function `import {urlParamsToObject} from '@okta/okta-auth-js/lib/oauthUtil';`  
After changing it to something like that `import {urlParamsToObject} from '@okta/okta-auth-js/lib/oauthUtil';` and install `@babel/runtime` package, it seems to be working fine.

**Comment by shuowu** [comment]
@rafakwolf Cool! 
files under `/lib` is in Typescript, files under `/cjs` is CommonJS modules, which can be consumed by node directly. 

The issue before you had may related to your ts config, which looks like did not transpile the `.ts` files from `auth-js` module (from node_modules) properly.

**Comment by rafakwolf** [comment]
> @rafakwolf Cool!
> files under `/lib` is in Typescript, files under `/cjs` is CommonJS modules, which can be consumed by node directly.
> 
> The issue before you had may related to your ts config, which looks like did not transpile the `.ts` files from `auth-js` module (from node_modules) properly.

I see, yeah I'm not transpiling files from `auth-js`.
Btw, is it possible to access the function `urlParamsToObject` from the dist folder?

**Comment by rafakwolf** [comment]
Also, why I need to install `@babel/runtime` package? Shouldn't that package be an `okta-authjs` dependency?

**Comment by shuowu** [comment]
The functions in `oauthUtil` module are designed for the client-side to use (same for build files under /dist) but looks like you want to use it in your server-side code, which means the interface will not be well supported and documented if anything changed in the future.

The server bundle is exposed from https://github.com/okta/okta-auth-js/blob/master/package.json#L9, but it doesn't have urlParamsToObject directly exposed. 

Per
>  why I need to install @babel/runtime package? Shouldn't that package be an okta-authjs dependency?

I still don't see why it happened for the server bundle, may need some investigation.

**Comment by rafakwolf** [comment]
> I still don't see why it happened for the server bundle, may need some investigation.
I've tried to uninstall it, but without installing it, the error I reported happens.

**Comment by kvbutler** [comment]
Guessing this package is not compatible with nodejs?

**Comment by shuowu** [comment]
@kvbutler `@okta/okta-auth-js` exposes server bundle for authN usage. The use case for `@okta/okta-sdk-nodejs` is to access OKTA's management API endpoints.

**Comment by kvbutler** [comment]
@shuowu thanks for the quick reply 👍 

I just realized that after reading through the @okta/okta-sdk-nodejs documentation, my use case is I need to fetch an id token for an e2e suite before any tests run.

I'm going to use the Resource Owner Password Flow now and make http requests (for the token) manually as is mentioned in the documentation https://developer.okta.com/docs/guides/implement-password/use-flow/.

**Comment by shubsaini09** [comment]
@shuowu I am not clear on this solution for this issue. I am getting the same error. 

I ran this command - npm i @okta/okta-auth-js --save and started my app with node app.js and got the below error

```
internal/modules/cjs/loader.js:834
  throw err;
  ^

Error: Cannot find module '@babel/runtime/helpers/interopRequireDefault'
Require stack:
- C:\Users\shubhams\Projects\okta\node_modules\@okta\okta-auth-js\cjs\server\index.js
- C:\Users\shubhams\Projects\okta\app.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:831:15)
    at Function.Module._load (internal/modules/cjs/loader.js:687:27)
    at Module.require (internal/modules/cjs/loader.js:903:19)
    at require (internal/modules/cjs/helpers.js:74:18)
    at Object.<anonymous> (C:\Users\shubhams\Projects\okta\node_modules\@okta\okta-auth-js\cjs\server\index.js:3:30)
    at Module._compile (internal/modules/cjs/loader.js:1015:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1035:10)
    at Module.load (internal/modules/cjs/loader.js:879:32)
    at Function.Module._load (internal/modules/cjs/loader.js:724:14)
    at Module.require (internal/modules/cjs/loader.js:903:19) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    'C:\\Users\\shubhams\\Projects\\okta\\node_modules\\@okta\\okta-auth-js\\cjs\\server\\index.js',
    'C:\\Users\\shubhams\\Projects\\okta\\app.js'
  ]
}
```

**Comment by shuowu** [comment]
@shubsaini09 After checking the `dependencies list` in okta-auth-js, seems like the [babel runtime](https://babeljs.io/docs/en/babel-runtime) is missing from there. For now, you can explicitly install it as a dependency in your app by running

```
npm install --save @babel/runtime
```
or 
```
yarn add @babel/runtime
```

Internal Ref: OKTA-354761

**Comment by shuowu** [comment]
Issue is fixed since 4.7.0
